### DISCUSSION TITLE
[Bug]: resolveStunIceServerIp Ignores Command Line AWS Region, Uses Environment Variable Instead

### DISCUSSION THREAD

**Discussion by PoshSylvester** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1971] [Q&A]
### Please confirm you have already done the following

- [X] I have searched the repository for related/existing bug reports
- [X] I have all the details the issue requires

### Please answer the following prompt

- [X] This issue is replicable using the unmodified sample application

### Describe the bug

The resolveStunIceServerIp function in the SDK always reads the AWS region from the environment variable, it does not use AWS region that is passed via the command line/terminal argument. This results in the default region being utilized if the DEFAULT_REGION_ENV_VAR is not explicitly set, despite an intended region being specified via the command line.

https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/f1073439213db4aa4b4b7cc1423609d920dd5489/src/source/PeerConnection/PeerConnection.c#L855-L858

### Expected Behavior

When the DEFAULT_REGION_ENV_VAR is not set, the AWS region specified via the command line arguments should be used.

This behavior can lead to unexpected defaults that may affect connectivity and performance, depending on the AWS region's configuration and latency differences.

### Current Behavior

The SDK defaults to using the AWS region specified in the environment variable (DEFAULT_REGION_ENV_VAR), even when a different region is passed via the command line. This behavior occurs consistently regardless of the command line input.

### Reproduction Steps

Check the code linked above

### WebRTC C SDK version being used

v1.10.0

### Compiler and Version used

Visual Studio

### Operating System and version

Windows 11

### Platform being used

Windows PC

**Comment by disa6302** [comment]
Hello @PoshSylvester ,

Command line argument is a sample specific construct. This functionality is within the SDK and it would not be possible to accomodate for different input modes of region to set this URL up. Our samples also do not have command line argument set up for the region.
